reconcile/utils/ocm_base_client.py

```diff
@@ -92,15 +92,15 @@ def get_paginated(
         # fetch pages
         records_seen = 0
         while True:
-            rs = self.get(api_path, params=params)
+            rs = self.get(api_path, params=params_copy)
             for item in rs.get("items", []):
                 yield item
             total_records = rs.get("total", 0)
             current_page = rs.get("page", 0)
             records_on_page = rs.get("size", len(rs.get("items", [])))
             records_seen += records_on_page
             if total_records > records_seen and (
-                max_pages and max_pages < current_page
+                not max_pages or max_pages > current_page
```

Comment:
when would `not max_pages` be true?

Reply:
You are right, the expression is still wrong. Guess a nested if/else or tenery is required.

Reply:
Is `max_pages` parameter used anywhere? In which case we want to early terminate pagination even not get full results?

Btw to keep pagination logic consistent with ocm cli https://github.com/openshift-online/ocm-cli/blob/bb3d95aee24b91436da8b5c2e2a46a681af5ff4a/cmd/ocm/list/cluster/cmd.go#L188C2-L215C1, can use if `records_on_page` is less than `max_page_size` to check if it's the last page. An edge case of that logic is it can send an additional request if the last page is a full page, but it's safer than check `total` as `total` value can change during pagination which can cause wrong termination point.

Reply:
Hm, how could the total value change? The old pagination logic worked as you proposed.

Reply:
@hemslo, aligned implementation. Please take a look